src/test/java/uk/gov/companieshouse/api/accounts/validation/DebtorsValidatorTest.java

```diff
@@ -0,0 +1,287 @@
+
+        when(transaction.getCompanyNumber()).thenReturn("12345");
+        when(environmentReader.getMandatoryString(any())).thenReturn("http://test-url");
+        when(restTemplate.getForObject(anyString(), any(Class.class))).thenReturn(companyProfile);
+
+        ReflectionTestUtils.setField(validator, "invalidNote", "invalid_note");
+
+        validator.isMultipleYearFiler(transaction);
+        errors = validator.validateDebtors(debtors, transaction);
+
+        Mockito.verify(restTemplate, Mockito.atLeastOnce()).getForObject(anyString(), any(Class.class));
+
+        assertTrue(errors.containsError(
+            new Error("invalid_note", PREVIOUS_TOTAL_PATH,
+                LocationType.JSON_PATH.getValue(),
+                ErrorType.VALIDATION.getType())));
+
+    }
+
+    @Test
+    @DisplayName("Tests Rest exception is thrown when company-profile not found")
+    public void testRestExceptionThrownWhenApiCallFails() {
+
+        addValidCurrentDebtors();
+
+        when(restTemplate.getForObject(anyString(), any(Class.class))).thenThrow(restClientException);
+
+        Executable executable = () -> {
+            validator.isMultipleYearFiler(transaction);
+        };
+        assertThrows(RestException.class, executable);
+    }
+
+
+    private void addValidCurrentDebtors() {
+
+        CurrentPeriod currentDebtors = new CurrentPeriod();
+        currentDebtors.setTradeDebtors(1L);
+        currentDebtors.setPrepaymentsAndAccruedIncome(2L);
+        currentDebtors.setGreaterThanOneYear(3L);
+        currentDebtors.setOtherDebtors(4L);
+        currentDebtors.setTotal(10L);
+        currentDebtors.setDetails("details");
+
+        debtors.setCurrentPeriod(currentDebtors);
+    }
+
+    @Test
+    @DisplayName("Tests current period incorrect total throws error")
+    public void testIncorrectCurrentTotal() {
+
+        CurrentPeriod currentDebtors = new CurrentPeriod();
+        currentDebtors.setTradeDebtors(1L);
+        currentDebtors.setPrepaymentsAndAccruedIncome(2L);
+        currentDebtors.setGreaterThanOneYear(3L);
+        currentDebtors.setOtherDebtors(4L);
+        currentDebtors.setTotal(100L);
+
+        debtors.setCurrentPeriod(currentDebtors);
+        ReflectionTestUtils.setField(validator, "incorrectTotal", "incorrect_total");
+
+        errors = validator.validateDebtors(debtors, transaction);
+
+        assertTrue(errors.containsError(
+            new Error("incorrect_total", CURRENT_TOTAL_PATH,
+                LocationType.JSON_PATH.getValue(),
+                ErrorType.VALIDATION.getType())));
+    }
+
+    @Test
+    @DisplayName("Tests current period missing total throws error")
+    public void testMissingCurrentTotal() {
+
+        CurrentPeriod currentDebtors = new CurrentPeriod();
+        currentDebtors.setTradeDebtors(1L);
+
+        debtors.setCurrentPeriod(currentDebtors);
+        ReflectionTestUtils.setField(validator, "invalidNote", "invalid_note");
```

Comment:
`invalidNote` and `invalid_note` are used a few times and could be extracted into constants:

`private static final String INVALID_NOTE_NAME = "invalidNote";`
`private static final String INVALID_NOTE_VALUE = "invalid_note";`